apisix/plugins/http-dubbo.lua

```diff
@@ -0,0 +1,242 @@
+        },
+        serialization_header_key = {
+            type = "string"
+        },
+        serialized = {
+            type = "boolean",
+            default = false
+        }
+    },
+    required = { "service_name", "method", "params_type_desc" },
+}
+
+local _M = {
+    version = 0.1,
+    priority = 0,
+    name = plugin_name,
+    schema = schema,
+}
+
+function _M.check_schema(conf)
+    return core.schema.check(schema, conf)
+end
+
+local function str_int32(int)
+    return char(band(rshift(int, 24), 0xff),
+            band(rshift(int, 16), 0xff),
+            band(rshift(int, 8), 0xff),
+            band(int, 0xff))
+end
+
+local function parse_dubbo_header(header)
+    for i = 1, 16 do
+        local currentByte = header:byte(i)
+        if not currentByte then
+            return nil
+        end
+    end
+
+    local magic_number = str_format("%04x", header:byte(1) * 256 + header:byte(2))
+    local message_flag = header:byte(3)
+    local status = header:byte(4)
+    local request_id = 0
+    for i = 5, 12 do
+        request_id = request_id * 256 + header:byte(i)
+    end
+
+    local byte13Val = header:byte(13) * 256 * 256 * 256
+    local byte14Val = header:byte(14) * 256 * 256
+    local data_length = byte13Val + byte14Val + header:byte(15) * 256 + header:byte(16)
+
+    local is_request = bit.band(bit.rshift(message_flag, 7), 0x01) == 1 and 1 or 0
+    local is_two_way = bit.band(bit.rshift(message_flag, 6), 0x01) == 1 and 1 or 0
+    local is_event = bit.band(bit.rshift(message_flag, 5), 0x01) == 1 and 1 or 0
+
+    return {
+        magic_number = magic_number,
+        message_flag = message_flag,
+        is_request = is_request,
+        is_two_way = is_two_way,
+        is_event = is_event,
+        status = status,
+        request_id = request_id,
+        data_length = data_length
+    }
+end
+
+local function string_to_json_string(str)
+    local result = "\""
+    for i = 1, #str do
+        local byte = core.string.sub(str, i, i)
+        if byte == "\\" then
+            result = result .. "\\\\"
+        elseif byte == "\n" then
+            result = result .. "\\n"
+        elseif byte == "\t" then
+            result = result .. "\\t"
+        elseif byte == "\r" then
+            result = result .. "\\r"
+        elseif byte == "\b" then
+            result = result .. "\\b"
+        elseif byte == "\f" then
+            result = result .. "\\f"
+        elseif byte == "\"" then
+            result = result .. "\\\""
+        else
+            result = result .. byte
+        end
+    end
+    return result .. "\""
+end
+
+local function get_dubbo_request(conf, ctx)
+    -- use dubbo and fastjson
+    local first_byte4 = "\xda\xbb\xc6\x00"
+
+    local requestId = "\x00\x00\x00\x00\x00\x00\x00\x01"
+    local version = "\"2.0.2\"\n"
+    local service = "\"" .. conf.service_name .. "\"" .. "\n"
+
+    local service_version
+    if not conf.service_version then
+        service_version = "0.0.0"
```

Comment:
The default configuration should be used here instead of hard-coded

Reply:
fixed